std/os/path.zig

```diff
@@ -30,42 +31,37 @@ pub fn isSep(byte: u8) bool {
     }
 }
 
-/// Naively combines a series of paths with the native path seperator.
-/// Allocates memory for the result, which must be freed by the caller.
 pub fn join(allocator: &Allocator, paths: ...) ![]u8 {
-    if (is_windows) {
-        return joinWindows(allocator, paths);
-    } else {
-        return joinPosix(allocator, paths);
-    }
+    // Currently a work around for the expansion to allow string separators 
```

Comment:
Just converting a character into a string since the new join takes a string separator since I found it rarer to want a character than a string i.e. ", " is very common.  If you know of a way to convert a character into a string without this little workaround then I would be overjoyed :).

Reply:
Currently, I've decided to have a function to handle that, it literally does just convert it into an array like `[]u8 { sep }` so functionally its the same but atleast its a bit nicer :).  So you can just call `string.joinCharSep()` and give a single character sep rather than a string :).